src/js_results.hpp

```diff
@@ -101,8 +101,11 @@ typename T::Object ResultsClass<T>::create_instance(ContextType ctx, realm::Resu
 }
 
 template<typename T>
-typename T::Object ResultsClass<T>::create_instance(ContextType ctx, SharedRealm realm, const ObjectSchema &object_schema) {
-    auto table = ObjectStore::table_for_object_type(realm->read_group(), object_schema.name);
+typename T::Object ResultsClass<T>::create_instance(ContextType ctx, SharedRealm realm, const std::string &object_type) {
```

Comment:
Shouldn't we just fix this in object store when creating a Results object? We have run into this issue multiple times and should fix it at the root of the problem to prevent it from happening again in the future.